Trio/Sources/Modules/Treatments/TreatmentsStateModel.swift

```diff
@@ -474,14 +475,19 @@ extension Treatments {
                     }
                     await apsManager.enactBolus(amount: maxAmount, isSMB: false, callback: nil)
                 } else {
-                    print("authentication failed")
+                    debug(.bolusState, "Authentication failed")
+                    throw UnlockError(error: LAError(.authenticationFailed))
                 }
             } catch {
-                print("authentication error for pump bolus: \(error.localizedDescription)")
+                debug(.bolusState, "Authentication error for pump bolus: \(error)")
                 await MainActor.run {
                     self.isAwaitingDeterminationResult = false
                     self.showDeterminationFailureAlert = true
-                    self.determinationFailureMessage = error.localizedDescription
+                    self
+                        .determinationFailureMessage =
+                        String(
+                            localized: "Authentication failed. Please ensure you have configured a Passcode for your iPhone under iOS Settings > Face ID & Code."
```

Comment:
Do we need to have separate strings for authentication failure, passcode not set, and maybe a generic error? The exception should have the details:

https://developer.apple.com/documentation/localauthentication/laerror-swift.struct/code

Reply:
I‘ll lock into adding a switch case here and adjust the error message response.